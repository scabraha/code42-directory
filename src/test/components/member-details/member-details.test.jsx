import React from 'react';
import renderer from 'react-test-renderer';
import MemberDetails from '../../../components/member-details/member-details';
import RepoView from '../../../components/repo-view';

jest.mock('../../../components/repo-view', () => 'RepoView');

describe('Member Details', () => {
  it('Renders the view with all props and ensures they show', () => {
    const tree = renderer.create(
      <MemberDetails
        joinDate='Today'
        location='USA'
        email='test@domain.com'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with no email and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberDetails
        joinDate='Today'
        location='USA'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with no location and ensures it renders correctly', () => {
    const tree = renderer.create(
      <MemberDetails
        joinDate='Today'
        email='test@domain.com'
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with no location or email and ensures it renders correctly', () => {
    const tree = renderer.create(<MemberDetails joinDate='Today' />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

