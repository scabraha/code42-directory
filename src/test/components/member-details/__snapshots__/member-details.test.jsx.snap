// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Details Renders the view with all props and ensures they show 1`] = `
<div
  className="member-details"
>
  <div
    className="member-details-info md-body-2"
  >
    <span>
      <a
        href="mailto://test@domain.com"
      >
        test@domain.com
      </a>
    </span>
    <div>
      Joined: 
      Invalid Date
    </div>
    <div>
      <i
        className="material-icons"
      >
        
      </i>
      USA
    </div>
  </div>
  <RepoView />
</div>
`;

exports[`Member Details Renders the view with no email and ensures it renders correctly 1`] = `
<div
  className="member-details"
>
  <div
    className="member-details-info md-body-2"
  >
    <div>
      Joined: 
      Invalid Date
    </div>
    <div>
      <i
        className="material-icons"
      >
        
      </i>
      USA
    </div>
  </div>
  <RepoView />
</div>
`;

exports[`Member Details Renders the view with no location and ensures it renders correctly 1`] = `
<div
  className="member-details"
>
  <div
    className="member-details-info md-body-2"
  >
    <span>
      <a
        href="mailto://test@domain.com"
      >
        test@domain.com
      </a>
    </span>
    <div>
      Joined: 
      Invalid Date
    </div>
  </div>
  <RepoView />
</div>
`;

exports[`Member Details Renders the view with no location or email and ensures it renders correctly 1`] = `
<div
  className="member-details"
>
  <div
    className="member-details-info md-body-2"
  >
    <div>
      Joined: 
      Invalid Date
    </div>
  </div>
  <RepoView />
</div>
`;
